File: unpack.rb
require 'msgpack'
require 'benchmark'

filename = ARGV.shift

File.open(filename) do |f|
  unpacker = MessagePack::Unpacker.new(f)
  Benchmark.bm do |x|
    x.report {
      unpacker.each do |obj|
      end
    }
  end
end

File: unpack.py
import msgpack
import io
import sys
from benchmarker import Benchmarker

with Benchmarker() as bench:
    @bench('msgpack')
    def _(bm):
        in_ch = io.open(sys.argv[1], mode='rb')
        unpacker = msgpack.Unpacker(in_ch, encoding=str("utf-8"))
        for row in unpacker:
            pass
